### PR TITLE
objectives: fixed dtypes and improved vlmop2

### PR DESCRIPTION
**Pull request by hstojic**

### CODE REVIEW

File: trieste/objectives/multi_objectives.py [hunk 1]
```diff
@@ -55,39 +55,49 @@ class MultiObjectiveTestProblem(ObjectiveTestProblem):
...
+    :param x: The points at which to evaluate the function, with shape [..., d].
+    :param d: The dimensionality of the synthetic function.
     :return: The function values at ``x``, with shape [..., 2].
     :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
     """
-    tf.debugging.assert_shapes([(x, (..., 2))], message="vlmop2 only allow 2d input")
-    transl = 1 / math.sqrt(2.0)
+    tf.debugging.assert_shapes(
+        [(x, (..., d))],
+        message=f"input x dim: {x.shape[-1]} is not align with pre-specified dim: {d}",
```
Comment (uri-granta) [inline trieste/objectives/multi_objectives.py]:
```suggestion
        message=f"input x dim: {x.shape[-1]} does not align with pre-specified dim: {d}",
```

File: trieste/objectives/multi_objectives.py [hunk 2]
```diff
@@ -146,15 +156,21 @@ def g(xM: TensorType) -> TensorType:
 def DTLZ1(input_dim: int, num_objective: int) -> MultiObjectiveTestProblem:
     """
     The DTLZ1 problem, the idea pareto fronts lie on a linear hyper-plane.
-    See :cite:deb2002scalable for details.
+    See :cite:`deb2002scalable` for details.
+
+    :param input_dim: The input dimensionality of the synthetic function.
```
Comment (uri-granta) [inline trieste/objectives/multi_objectives.py:161]: good spot

File: trieste/objectives/multi_objectives.py [hunk 3]
```diff
@@ -55,39 +55,49 @@ class MultiObjectiveTestProblem(ObjectiveTestProblem):
...
     y1 = 1 - tf.exp(-1 * tf.reduce_sum((x - transl) ** 2, axis=-1))
     y2 = 1 - tf.exp(-1 * tf.reduce_sum((x + transl) ** 2, axis=-1))
     return tf.stack([y1, y2], axis=-1)
 
 
-def vlmop2_gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
-    tf.debugging.assert_greater(n, 0)
-    _x = tf.linspace([-1 / tf.sqrt(2.0)], [1 / tf.sqrt(2.0)], n)
-    return vlmop2(tf.concat([_x, _x], axis=1))
+def VLMOP2(input_dim: int) -> MultiObjectiveTestProblem:
```
Comment (uri-granta) [inline trieste/objectives/multi_objectives.py:77]: I assume the 2 in VLMOP2 doesn't refer to the dimensions then?

**Review by uri-granta — APPROVED** [review]
Looks good, thanks!

File: trieste/objectives/multi_objectives.py (hunk 3 above)
Comment (hstojic) [reply trieste/objectives/multi_objectives.py:77]: it does, dimensionality of the search space... why do you say that?

**Pull request merged**